libs/partners/mongodb/langchain_mongodb/index.py

```diff
@@ -0,0 +1,164 @@
+import logging
+from typing import Any, Dict, List, Optional
+
+from pymongo.collection import Collection
+
+logger = logging.getLogger(__file__)
+
+
+def _create_index_definition(
+    dimensions: int,
+    path: str,
+    similarity: str,
+    filters: Optional[List[Dict[str, str]]],
+) -> Dict[str, Any]:
+    return {
+        "fields": [
+            {
+                "numDimensions": dimensions,
+                "path": path,
+                "similarity": similarity,
+                "type": "vector",
+            },
+            *(filters or []),
+        ]
+    }
+
+
+def ensure_vector_search_index(
+    collection: Collection,
+    index_name: str,
+    dimensions: int,
+    path: str,
+    similarity: str,
+    filters: List[Dict[str, str]],
+    update: bool = False,
+) -> None:
+    """Creates a MongoDB Atlas Vector Search Index
+
+    Args:
+        collection (Collection): MongoDB Collection
+        index_name (str): Name of Index
+        dimensions (int): Number of dimensions in embedding
+        path (str): field with vector embedding
+        similarity (str): The similarity score used for the index
+        filters (List[Dict[str, str]]): additional filters for index definition.
+        update (bool, optional): Update Existing Search index. Defaults to False.
+    """
+    index_exists = bool(list(collection.list_search_indexes(index_name)))
+
+    if update:
+        if index_exists:
+            update_vector_search_index(
+                collection,
+                index_name,
+                dimensions,
+                path,
+                similarity,
+                filters,
+            )
+        else:
+            logger.info(
```

Comment:
I would expect this to raise an error.

Reply:
I've been on the fence about raising an error on this. I had assumed folks would often reboot the code. If it's going to raise an error, then I don't need to include any if statements.

Reply:
Removing the `index_exists` call. Placing the OperationFailure message here for posterity: 
```
pymongo.errors.OperationFailure: FTS index with name catapult cannot be found, full error: {'ok': 0.0, 'errmsg': 'FTS index with name catapult cannot be found', 'code': 27, 'codeName': 'IndexNotFound', '$clusterTime': {'clusterTime': Timestamp(1710988571, 1), 'signature': {'hash': b'\x93\x00"\x88\t\xea\xcbyUN\xc4ike\xde\xef<\xd4\xe3Q', 'keyId': 7292037883655880706}}, 'operationTime': Timestamp(1710988571, 1)}
```